include/libaktualizr/types.h

```diff
@@ -0,0 +1,461 @@
+﻿#ifndef TYPES_H_
+#define TYPES_H_
+/** \file */
+
+#include <algorithm>
+#include <stdexcept>
+
+#include <json/json.h>
+#include <boost/filesystem.hpp>
+
+enum class ProvisionMode { kSharedCred = 0, kDeviceCred };
+enum class StorageType { kFileSystem = 0, kSqlite };
+
+/**
+ * @brief The BasedPath class
+ * Can represent an absolute or relative path, only readable through the BasePath::get() method.
+ *
+ * The intent is to avoid unintentional use of the "naked" relative path by
+ * mandating a base directory for each instantiation.
+ *
+ * @todo has to be moved into Utils namespace
```

Comment:
this syntax is a ```doxygen``` compliant, I know we use ```TODO``` but what the reason to keep this approach in our public headers!

Reply:
But is that "todo" more meaningful for the API users reading the documentation or for us, the developers that maintain it and whose burden it would be to actually put it in the `Utils` namespace?

More importantly, is it actually important for `BasedPath` to live in the `Utils` namespace? I'm not convinced it "has to be moved".